api/envoy/config/filter/http/grpc_stats/v2alpha/config.proto

```diff
@@ -20,6 +23,32 @@ message FilterConfig {
   // If true, the filter maintains a filter state object with the request and response message
   // counts.
   bool emit_filter_state = 1;
+
+  oneof per_method_stat_specifier {
+    // If set, specifies an allowlist of service/methods that will have individual stats
+    // emitted for them. Any call that does not match the allowlist will be counted
+    // in a stat with no method specifier: `cluster.<name>.grpc.*`.
+    api.v2.core.GrpcMethodList individual_method_stats_allowlist = 2;
+
+    // If set to true, emit stats for all service/method names.
+    //
+    // If set to false, emit stats for all service/message types to the same stats without including
+    // the service/method in the name, with prefix `cluster.<name>.grpc`. This can be useful if
+    // service/method granularity is not needed, or if each cluster only receives a single method.
+    //
+    // .. attention::
+    //   This option is only safe if all clients are trusted. If this option is enabled
+    //   with untrusted clients, the clients could cause unbounded growth in the number of stats in
+    //   Envoy, using unbounded memory and potentially slowing down stats pipelines.
+    //
+    // .. attention::
+    //   If neither `stats_allowlist` nor `stats_for_all_methods` is set, the behavior
+    //   will default to `stats_for_all_methods=true`. This default value is deprecated, and in a
+    //   future release, if neither field is set, it will default to `stats_for_all_methods=false`
```

Comment:
In general, the idea of changing behaviors across APIs in Envoy is a bad one, but for defaults I think we can allow it, since management servers that desire stability can set explicit values.

Reply:
I agree; I don't really like this kind of behavior change. But on the other hand, I don't like default-unsafe behavior either.

In this case, the change is only changing which stats are emitted, so if it went unnoticed in a deployment, it may cause alerts or lack of metrics, but it would not disrupt the flow of requests.

If we want to undo this, and leave the default unsafe with the `attention` block, I could be talked into that.

Reply:
Yeah, as I said, I think this is an exception that we can make as API shepherds, since I don't think management servers should rely on default behavior as stable. I'll put a PR out to clarify this for future.

Reply:
Actually, this is already covered in the last paragraph of https://github.com/envoyproxy/envoy/blob/master/api/API_VERSIONING.md#backwards-compatibility. PEBKAC.